### DISCUSSION TITLE
Improvement for Guided Drawing Inbetweener

### DISCUSSION THREAD

**Discussion by Abuganim** [opentoonz/opentoonz#6538] [Ideas]
Hi, I'm an animator and been learning OT for a while, I find it better with each iteration,
Tho I would like to propose this request for making an improvement for the Guided Drawing inbetweener feature.

The algoritm for interpolation seems to mess with the **strokes's shapes and
positions so they Dont Retain the shape nor the conections** from key to key

I think perhaps the algoritm could be fixed so the strokes **could retain some degree of conection
with the near strokes  and you could determine the ARCs in which the strokes's inbetweens would travel** 
 from one key stroke to another, or for several ones at the same time in a drawing.

Here I leave simple example why I consider  the inbetweener is not yet ready for making
consistent inbetweens.

![Gif_OpenToonz_2](https://user-images.githubusercontent.com/66708840/84292608-37106500-ab0c-11ea-9411-c89bc1af0ce9.gif)

**Comment by ghost** [comment]
I also think that it would be a very big step in this kind of workflow if the spacing were taken into account between two keys. This would make OT much more competitive against Cacani. A great time saver. But surely something very difficult to correct.

**Comment by melieconiek** [comment]
OpenToonz stoke inbtweening is already pretty fantastic, though not infoulable. I would also really like custom easing profiles--the timing chart in Cacani is a really great implementation of this.

@Abuganim This isn't a complete solution for you--but I've found that splitting the parts into different levels and translating them with the animate tool instead of letting OT's auto-tween handle the translation gets much better results. You can even set it up like a cutout animation. When you're done, you can flatten it all down into a single level using the merge command. You have a lot more control over everything when you work this way.

1. Use the animate tool to translate your drawings to rough position of the next extreme.
2. Create a new drawing and draw your extreme.
3. Create your inbetween frames and use the auto-tween.
4. Adjust your arc and easing using the animate tool and make any corrections necessary to the tween.
5. Create the necessary drawings in your master level and merge your completed tweens down. Do any additional cleanup needed.

**Comment by Abuganim** [comment]
@openanim  you're right, i also think the predetermined options of ease in ease out are really not sufficient, there should be complete control over the spacing, since its the escence of animation.

**Comment by Abuganim** [comment]
@melieconiek Thats a very interesting work around, tho i think it still doesn't resolve the problem, since you'd have to make mixup of frame by frame and cut out animation and managing motions and several levels and the pipeline gets messy and confusing.

But if the interpolation itself could be enhanced, you could easily use a single layer for achieving good results even in complex drawings like in anime which are very detailed. 

But right  now, in the state it is the autoinbetweener feature, it has a lot of trouble for successfully acomplishing  even simple interpolations as the one i posted.

I really like OpenToonz, it is a fantastic tool, but I consider the autoinbetweener feature has still a way to go to be really efficient and functional right now. And it'll be tremendous for OT having this feature improved and make consistent inbetweens not only morphing ones.
